torch/custom_class_detail.h

```diff
@@ -51,6 +37,51 @@ struct pickle_factory<Get, Set, RetState(Self), NewInstance(ArgState)> {
   Set s;
 };
 
+template <typename Method>
+struct WrapMethod;
+
+template <typename R, typename CurrClass, typename... Args>
+struct WrapMethod<R (CurrClass::*)(Args...)> {
+  WrapMethod(R (CurrClass::*m)(Args...)) : m(std::move(m)) {}
+
+  R operator()(c10::intrusive_ptr<CurrClass> cur, Args... args) {
+    return c10::guts::invoke(m, *cur, args...);
```

Comment:
- you probably want to std::forward the args
- cur->(*m)(...) should work here and is easier to understand than a call to invoke().